Adadie/Fresh
# #37-CRUD

#### What does this PR do?
This PR will add firebase SDK to the contacts page, dashboard, signup page and login page. It will add email & password authentication to the user signup page, add a firestore reference to contact page and post article page. Add CRUD scripts for messages and posting articles.

#### Description of Task to be completed?

- [x] Add firebase SDK to the contacts page, dashboard, signup page and login page.
- [x] Add email & password authentication to the user signup page
- [x] Add a firestore reference to contact page and post article page.
- [x] Add validation on signup page
- [x] Add delete and update button on posts and messages.

#### How should this be manually tested?

- Clone this branch to your local repo [link](https://github.com/Junior-Adadie/Fresh.git)
- Open profile.html with your browser 
- Select the create post section
- Fill in the form and send
- Check if the posted article is in the articles section.
- Click the delete button and see if the article is deleted
- Enter the contact page
- Fill in the contact page form and send
- Check in the dashboard page if the message is displayed there or inspect with console to see if the data is sent

#### Any background context you want to provide?

Sometimes the Contacts page responds when you comment out the post blogs section in 

#### What are the relevant trello stories?

[Card 1](https://trello.com/c/rKgQjdRP/37-crud)
[Card 2](https://trello.com/c/S9Gqai9U/36-functions-for-sending-data-are-conflicting)
[Caard 3](https://trello.com/c/GR37M9ew/31-store-registation-info-on-firebase)
[Card 4](https://trello.com/c/53oZ2eEy/26-store-queries-posts-on-firestore)
[Card 5](https://trello.com/c/csutfegO/22-validate-forms)
#### Screenshots (if appropriate)
![Postart](https://user-images.githubusercontent.com/68109866/89763332-f7a9c900-dafa-11ea-832d-3336c3911af4.PNG)
![Posted](https://user-images.githubusercontent.com/68109866/89763345-fd071380-dafa-11ea-817c-8e218253040a.PNG)
